lib/wallaby/phantom/driver.ex

```diff
@@ -333,6 +333,135 @@ defmodule Wallaby.Phantom.Driver do
     end
   end
 
+  @doc """
+  Accept all JavaScript dialogs
+  """
+  def accept_dialogs(session) do
+    script = """
+    var page = this;
+    page.onAlert = function(msg) {}
+    page.onConfirm = function(msg) { return true; }
+    page.onPrompt = function(msg, defaultVal) { return defaultVal; }
+    return "ok";
+    """
+    {:ok, "ok"} = execute_phantom_script(session, script)
+  end
+
+  @doc """
+  Dismiss all JavaScript dialogs
+  """
+  def dismiss_dialogs(session) do
+    script = """
+    var page = this;
+    page.onAlert = function(msg) {}
+    page.onConfirm = function(msg) { return false; }
+    page.onPrompt = function(msg, defaultVal) { return null; }
+    return "ok";
+    """
+    {:ok, "ok"} = execute_phantom_script(session, script)
+  end
+
+  @doc """
+  Accept one alert triggered within `fun` and return the alert message.
+  """
+  def accept_alert(%Session{}=session, fun) do
+    script = """
+    var page = this;
+    page.__onAlertDefault = page.onAlert;
+    page.onAlert = function(msg) {
+      page.__alertMessage = msg;
+      page.onAlert = page.__onAlertDefault;
+    }
+    return "ok";
+    """
+    {:ok, "ok"} = execute_phantom_script(session, script)
+    fun.(session)
+    script = """
+    var page = this;
+    page.onAlert = page.__onAlertDefault;
+    return ["ok", page.__alertMessage];
+    """
+    {:ok, ["ok", message]} = execute_phantom_script(session, script)
+    message
+  end
+
+  @doc """
+  Accept one confirm triggered within `fun` and return the confirm message.
+  """
+  def accept_confirm(%Session{}=session, fun) do
+    handle_confirm(session, fun, true)
+  end
+
+  @doc """
+  Dismiss one confirm triggered within `fun` and return the confirm message.
+  """
+  def dismiss_confirm(%Session{}=session, fun) do
+    handle_confirm(session, fun, false)
+  end
+
+  defp handle_confirm(%Session{}=session, fun, return_value) do
+    script = """
+    var page = this, returnVal = arguments[0];
+    page.__onConfirmDefault = page.onConfirm;
+    page.onConfirm = function(msg) {
+      page.__confirmMessage = msg;
+      page.onConfirm = page.__onConfirmDefault;
+      return returnVal;
+    }
+    return "ok";
+    """
+    {:ok, "ok"} = execute_phantom_script(session, script, [return_value])
+    fun.(session)
+    script = """
+    var page = this;
+    page.onConfirm = page.__onConfirmDefault;
+    return ["ok", page.__confirmMessage];
+    """
+    {:ok, ["ok", message]} = execute_phantom_script(session, script)
+    message
```

Comment:
@PragTob @keathley I am not so sure any more that this is worth it. The initial idea was that we wouldn't have to "interrupt" the pipeline in order to get the message later, but this approach kind of fails to live up to the expectations. I was trying out the approach just now and it turns out that instead of writing the test case like this:

```elixir
test "accept window.alert and get message", %{page: page} do
  message = accept_alert page, fn(p) ->
    click(p, Query.link("Alert"))
  end
  result =
    page
    |> find(Query.css("#result"))
    |> Element.text()
  assert message == "This is an alert!"
  assert result == "Alert accepted"
end
```

I now have to write it like this, because the modified `session` needs to be stored before we can continue:

```elixir
test "accept window.alert and get message", %{page: page} do
  page =
    page
    |> accept_alert(fn(p) ->
      click(p, Query.link("Alert"))
    end)
  result =
    page
    |> find(Query.css("#result"))
    |> Element.text()
  assert last_dialog_message(page) == "This is an alert!"
  assert result == "Alert accepted"
end
```

I am honestly wondering if this is any better than before, in particular if this is worth the additional complexity, having to keep track of messages in the session struct. Basically, this only simplifies the case you're using the finer grained `accept_alert` etc functions, but are not interested in the message.

It would be great to have some feedback from you before I actually dig in and implement this.